pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -2437,6 +2438,42 @@ func (l *LibvirtDomainManager) GetFilesystems() []v1.VirtualMachineInstanceFileS
 	return fsList
 }
 
+func (l *LibvirtDomainManager) GetScreenshot(vmi *v1.VirtualMachineInstance) (*v1.ScreenshotResponse, error) {
```

Comment:
@victortoso @0xFelix did you consider rate limiting? 
Isn't this a concern?

Reply:
@vladikr good point. I did think about rate limiting, on this thread https://github.com/kubevirt/kubevirt/pull/15238#discussion_r2228084608

The code that this replaces had nothing like that, that's the only reason I'm suggesting to make it a follow-up but if you think we should do it in this PR let me know.

Also (because your snip is from virt-launcher), I'd expect that any sort of rate-limiting would be on virt-api side as in virt-launcher would already be late. Perhaps on the virt-launcher side we should only worry about memory increase, but I'm not sure if we have a common piece of code in the launcher to validate that we are on the limit or not of the memory allocation.